includes/model/class-user.php

```diff
@@ -146,7 +176,30 @@ public function get_icon() {
 		);
 	}
 
+	public function save_icon( $icon ) {
+		$maybe_id = (int) $icon;
+		if ( $maybe_id ) {
+			$image = wp_get_attachment_image( $maybe_id, 'full' );
+			if ( ! $image ) {
+				return false;
+			}
+			$icon = wp_get_attachment_url( $maybe_id );
+		}
+
+		$option_name = 'activitypub_user_icon_' . $this->_id;
```

Comment:
For signatures we use `$user->user_login` instead, because the ID can change on migrations!?!

Reply:
That seems like a pretty good idea. For dotcom Simple -> Atomic migrations we do maintain the User ID although that may not always be the case when we do something like a WXR Import

Reply:
But we should be consistent! I am already having a hard time with this login key on the deletion feature. If we decide to use the id instead then I will migrate that stuff!